### DISCUSSION TITLE
How to use setLegend

### DISCUSSION THREAD

**Discussion by cw-dvr8** [acorg/Racmacs#174] [Q&A] [answered]
I'm mostly a Python programmer getting started with R.  I used Racmacs to do an antigenic cartography plot, but can't figure out how to get the setLegend function to work.  I need a little more help than what's in the docs, and I haven't been able to find any example code anywhere online to play with.  Does anyone have a sample of code that uses setLegend that I can look at?  Thanks!

**Comment by drserajames** [comment, marked as answer]
Does this help? I've not used the function before, so there might be a smarter way to do things. If you want a legend on a plot, then you can use the R `legend` function.

```
library(Racmacs)

map <- read.acmap(system.file("extdata/h3map2004.ace", package = "Racmacs"))

legend_map <- setLegend(map, 
  c("HK68", "EN72", "VI75", "TX77", "BK79", "SI87", "BE89", "BE92", "WU95", "SY97", "FU02"),
  fill=unique(agFill(map)))

view(legend_map)
```

**Reply by cw-dvr8** [reply]
Thanks for this!  I hadn't realized that setLegend was only applicable for the view() function, so I used the legend() function.  It gave me a bit of an odd placement because of the padding and I'll need to eventually figure out the xlim and ylim parameters to hopefully mitigate this, but it will do for now.  Thanks again!

**Reply by drserajames** [reply]
Setting the margins to zero (or some other value) might fix your legend placement issue.

```
par(mar=rep(0,4))
plot(map)
legend("topright", c("something", "somethingelse"))
```

**Comment by shwilks** [comment]
You beat me to it Sarah, thanks for taking the time to post your question here for others to see, here's some other example code demonstrating the same thing:

```R
map <- setLegend(
  map = map,
  legend = c(
    "egg-passaged viruses",
    "cell-passaged viruses"
  ),
  fill = c(
    "red",
    "blue"
  )
)
view(map)
```
